### PR TITLE
existing-payment-options : belated minor refactors

### PR DESCRIPTION
**Pull request by twrichards**
some minor refactors following comments on https://github.com/guardian/support-frontend/pull/1673

_Somehow managed to forget to create a PR for these_

### CODE REVIEW

File: support-frontend/assets/helpers/existingPaymentMethods/existingPaymentMethods.js [hunk 1]
```diff
@@ -17,24 +17,23 @@ export type ExistingPaymentMethodSubscription = {
 
 type ExistingPaymentType = 'Card' | 'DirectDebit';
 
-export type PartialExistingPaymentMethod = {
-  paymentType: ExistingPaymentType;
+export type NotRecentlySignedInExistingPaymentMethod = {
```
Comment (rupertbates) [inline support-frontend/assets/helpers/existingPaymentMethods/existingPaymentMethods.js]: That's a real mouthful, is there anyway we could come up with something more succinct?

Comment (twrichards) [reply support-frontend/assets/helpers/existingPaymentMethods/existingPaymentMethods.js]: i agree, but it is explicit - it is in reaction to https://github.com/guardian/support-frontend/pull/1673#pullrequestreview-219572132 - the shorter name before was more ambiguous

File: support-frontend/assets/helpers/existingPaymentMethods/existingPaymentMethods.js [hunk 2]
```diff
@@ -17,24 +17,23 @@ export type ExistingPaymentMethodSubscription = {
...
-  paymentType: ExistingPaymentType
 };
 
-export type ExistingPaymentMethod = PartialExistingPaymentMethod | FullDetailExistingPaymentMethod;
+export type ExistingPaymentMethod = NotRecentlySignedInExistingPaymentMethod | RecentlySignedInExistingPaymentMethod;
 
 // ----- Functions ----- //
 
-function isFullDetailExistingPaymentMethod(existingPaymentMethod: ExistingPaymentMethod): boolean %checks {
+function isUsableExistingPaymentMethod(existingPaymentMethod: ExistingPaymentMethod): boolean %checks {
```
Comment (rupertbates) [inline support-frontend/assets/helpers/existingPaymentMethods/existingPaymentMethods.js]: I hadn't seen %checks or the double bang to return a boolean before, interesting!

File: support-frontend/assets/pages/new-contributions-landing/components/PaymentMethodSelector.jsx [hunk 3]
```diff
@@ -99,6 +99,10 @@ function getPaymentMethodLogo(paymentMethod: PaymentMethod) {
   }
 }
 
+function isRecurringContribution(contributionType: ContributionType) {
```
Comment (rupertbates) [inline support-frontend/assets/pages/new-contributions-landing/components/PaymentMethodSelector.jsx]: This function already exists in contributions.js line 32

**Pull request merged**